### PR TITLE
[openshift-base] is namespace deleted

### PR DESCRIPTION
**Pull request by maorfr**
instead of duplicating `namespace_info.get("delete")`

### CODE REVIEW

File: reconcile/test/test_openshift_base.py [hunk 1]
```diff
@@ -645,3 +645,18 @@ def test_determine_user_keys_for_access_not_implemented():
     auth = {"service": "not-implemented"}
     with pytest.raises(NotImplementedError):
         sut.determine_user_keys_for_access("cluster-name", [auth])
+
```
Comment (geoberle) [inline reconcile/test/test_openshift_base.py:648]: add another one for `ns = {"delete": None}`

File: reconcile/openshift_base.py [hunk 2]
```diff
@@ -1004,3 +1004,7 @@ def determine_user_keys_for_access(
                 f"[{cluster_name}] auth service not implemented: {service}"
             )
     return user_keys
+
+
+def is_namespace_deleted(namespace_info: Mapping) -> bool:
```
Comment (maorfr) [inline reconcile/openshift_base.py:1009]: using `Mapping` for consistency with https://github.com/app-sre/qontract-reconcile/blob/8bca088d7bd0dcfc4ab437a6fc272e80be2baa05/reconcile/openshift_base.py#L72-L73

File: reconcile/test/test_openshift_base.py (hunk 1 above)
Comment (maorfr) [reply reconcile/test/test_openshift_base.py:648]: https://github.com/app-sre/qontract-reconcile/pull/3010/commits/d79b6a4647c415eab7a929f040a8c6d0728efe58

**Review by geoberle — APPROVED** [review]
lgtm

**Pull request merged**